From: Joonwon Kang <joonwonkang@google.com>
Subject: percpu: fix wrong chunk hints update
Date: Wed, 13 May 2026 08:51:13 +0000

Chunk end offset was set to a block end offset, which could prevent chunk
hints from being updated correctly.  It was observed that the chunk free
size gets minus or shorter than the actual free size due to this.  This
commit fixes it.

Link: https://lore.kernel.org/20260513085117.1024175-1-joonwonkang@google.com
Fixes: 92c14cab4326 ("percpu: convert chunk hints to be based on pcpu_block_md")
Signed-off-by: Joonwon Kang <joonwonkang@google.com>
Reviewed-by: Dennis Zhou <dennis@kernel.org>
Cc: Tejun Heo <tj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/percpu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/percpu.c~percpu-fix-wrong-chunk-hints-update
+++ a/mm/percpu.c
@@ -1054,7 +1054,7 @@ static void pcpu_block_update_hint_free(
 	else
 		pcpu_block_update(&chunk->chunk_md,
 				  pcpu_block_off_to_off(s_index, start),
-				  end);
+				  pcpu_block_off_to_off(e_index, end));
 }
 
 /**
_
